graphql-dgs-client/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/GraphQLQueryRequest.kt

```diff
@@ -63,10 +57,41 @@ class GraphQLQueryRequest(private val query: GraphQLQuery, private val projectio
         if (projection is BaseSubProjectionNode<*, *>) {
             builder.append(projection.root().toString())
         } else if (projection != null) {
+
             builder.append(projection.toString())
         }
 
         builder.append(" }")
         return builder.toString()
     }
 }
+
+class InputValueSerializer(private val scalars: Map<Class<*>, Coercing<*, *>> = emptyMap()) {
+    fun serialize(input: Any?): String? {
+        if (input == null) {
+            return null
+        }
+
+        val type = input::class.java
+
+        return if (scalars.contains(type)) {
+            """"${scalars[type]!!.serialize(input)}""""
+        } else if (type.isPrimitive || type.isAssignableFrom(java.lang.Integer::class.java) || type.isAssignableFrom(java.lang.Long::class.java) || type.isAssignableFrom(java.lang.Double::class.java) || type.isAssignableFrom(java.lang.Float::class.java) || type.isAssignableFrom(java.lang.Boolean::class.java) || type.isAssignableFrom(java.lang.Short::class.java) || type.isAssignableFrom(java.lang.Byte::class.java) || type.isEnum) {
+            input.toString()
+        } else if (type == String::class.java) {
+            """"$input""""
+        } else if (input is List<*>) {
+            """[${input.filterNotNull().joinToString(", ") { listItem -> serialize(listItem) ?: "" }}]"""
+        } else {
+            input.javaClass.declaredFields.filter { !it.type::class.isCompanion }.map {
```

Comment:
When I tried this I was getting null values and narrowed it down to here -- unfortunately declaredFields does not include inherited fields, so this breaks extended objects.  A quick trip to stackoverflow shows some solutions:

https://stackoverflow.com/questions/1042798/retrieving-the-inherited-attribute-names-values-using-java-reflection

Reply:
Good point, I've added support for extended objects now as well.